web/packages/teleport/src/Discover/useDiscover.tsx

```diff
@@ -66,25 +66,44 @@ type CustomEventInput = {
   eventName?: DiscoverEvent;
   eventResourceName?: DiscoverEventResource;
   autoDiscoverResourcesCount?: number;
+  selectedResourcesCount?: number;
 };
 
 type DiscoverProviderProps = {
   // mockCtx used for testing purposes.
   mockCtx?: DiscoverContextState;
 };
 
+// DiscoverUrlLocState define fields to preserve state between
+// react routes (eg. in RDS database flow, it is required of user
+// to create a AWS OIDC integration which requires changing route
+// and then coming back to resume the flow.)
+export type DiscoverUrlLocState = {
```

Comment:
I think we should change this to DiscoverUrlLocationState, just to be more clear. When we are working with it and setting it above, it's called `locationState` so makes more sense to keep it that way.